lib/puppet/provider/group/gpasswd.rb

```diff
@@ -0,0 +1,75 @@
+require 'puppet/provider/group/groupadd'
+
+Puppet::Type.type(:group).provide :gpasswd, :parent => Puppet::Type::Group::ProviderGroupadd do
+  require 'set'
+  require 'shellwords'
+
+  desc <<-EOM
+    Group management via `gpasswd`. This allows for local group
+    management when the users exist in a remote system.
+  EOM
+
+  commands  :addmember => 'gpasswd',
+            :delmember => 'gpasswd'
+
+  has_feature :manages_members unless %w{HP-UX Solaris}.include? Facter.value(:operatingsystem)
```

Comment:
I'm more asking the question if there are systems that have `gpasswd`, but are incapable of managing `members`? I'm assuming there are no systems that fall into that category, and we can remove the `unless %w{HP...}` part?

Note the `manages_members` provider feature won't determine suitability of the provider in general. It just only restricts when you can specify the desired list of `members`.

If you want to ensure the provider is only suitable on Linuxes that also have `gpasswd`, you'll want to add an explicit `confine` for that.